libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_mqtt_serialize.c

```diff
@@ -85,575 +85,704 @@ static inline uint16_t _getNumPublishParams( const MQTTBLEPublishInfo_t * const
-        error = IOT_BLE_MESG_ENCODER.appendKeyValue( &pubAckMap, IOT_BLE_MQTT_MSG_TYPE, data );
+        status = cbor_encode_text_string( &mapEncoder, IOT_BLE_MQTT_MESSAGE_ID, strlen( IOT_BLE_MQTT_MESSAGE_ID ) );
+
+        if( SERIALIZER_STATUS( status, pBuffer ) == true )
+        {
+            status = cbor_encode_int( &mapEncoder, packetIdentifier );
+        }
     }
 
-    if( _IS_VALID_SERIALIZER_RET( error, pBuffer ) )
+    if( SERIALIZER_STATUS( status, pBuffer ) == true )
     {
-        data.type = IOT_SERIALIZER_SCALAR_SIGNED_INT;
-        data.value.u.signedInt = packetIdentifier;
-        error = IOT_BLE_MESG_ENCODER.appendKeyValue( &pubAckMap, IOT_BLE_MQTT_MESSAGE_ID, data );
+        status = cbor_encoder_close_container( &encoder, &mapEncoder );
     }
 
-    if( _IS_VALID_SERIALIZER_RET( error, pBuffer ) )
+    if( pBuffer == NULL )
     {
-        error = IOT_BLE_MESG_ENCODER.closeContainer( &encoderObj, &pubAckMap );
+        *pLength = cbor_encoder_get_extra_bytes_needed( &encoder );
     }
-
-    if( _IS_VALID_SERIALIZER_RET( error, pBuffer ) )
+    else
     {
-        if( pBuffer == NULL )
-        {
-            *pSize = IOT_BLE_MESG_ENCODER.getExtraBufferSizeNeeded( &encoderObj );
-        }
-        else
-        {
-            *pSize = IOT_BLE_MESG_ENCODER.getEncodedSize( &encoderObj, pBuffer );
-        }
+        *pLength = cbor_encoder_get_buffer_size( &encoder, pBuffer );
+    }
 
-        IOT_BLE_MESG_ENCODER.destroy( &encoderObj );
-        error = IOT_SERIALIZER_SUCCESS;
+    if( SERIALIZER_STATUS( status, pBuffer ) == true )
+    {
+        result = CborNoError;
+    }
+    else
+    {
+        result = status;
     }
 
-    return error;
+    return result;
 }
 
 
-static IotSerializerError_t _serializeSubscribe( const MQTTBLESubscribeInfo_t * const pSubscriptionList,
-                                                 size_t subscriptionCount,
-                                                 uint8_t * const pBuffer,
-                                                 size_t * const pSize,
-                                                 uint16_t packetIdentifier )
+static CborError prvSerializeSubscribe( const MQTTBLESubscribeInfo_t * const pSubscriptionList,
+                                        size_t subscriptionCount,
+                                        uint8_t * const pBuffer,
+                                        size_t * const pLength,
+                                        uint16_t packetIdentifier )
 {
-    IotSerializerError_t error = IOT_SERIALIZER_SUCCESS;
-    IotSerializerEncoderObject_t encoderObj = IOT_SERIALIZER_ENCODER_CONTAINER_INITIALIZER_STREAM;
-    IotSerializerEncoderObject_t subscribeMap = IOT_SERIALIZER_ENCODER_CONTAINER_INITIALIZER_MAP;
-    IotSerializerEncoderObject_t subscriptionArray = IOT_SERIALIZER_ENCODER_CONTAINER_INITIALIZER_ARRAY;
-    IotSerializerScalarData_t data = { 0 };
-    uint16_t idx;
+    size_t length = *pLength;
+    CborError status, result;
```

Comment:
Maintainability: status values should always be initialized to a value meaning _error_ or _failure_.